### ISSUE TITLE
Feature: Allow/Block (only) once

### ISSUE THREAD

**Issue by fusionneur** [henrypp/simplewall#1150]
There should be a feature to allow/block access only once - for that specific PID, that specific IP and that specific port only until the connection is closed or reattempted again.

This is useful when trying to discover which are the ips an app is connecting to - in a granular way (while checking DNS name) by blocking and potentially unblocking the ips to discover whether blocking them is breaking any functionality in the app or that unblocking it restores the functionality.

One example would be for games, where nowadays they attempt a lot of connections, and it is difficult to distinguish between telemetry and needed endpoints for the game to work. For example, I run the game, then, as it attempts connections, i block the telemetry endpoints and once I figure out that there are no other telemetry endpoints attempted, I eventually allow full HTTPS + UDP (let's say).

**Comment by fusionneur** [comment]
The behaviour i suggest is similar to ESET EIS's firewall in interactive mode. It is quite hard to explain until experiencing this feature and its benefits.

**Comment by fusionneur** [comment]
Nowadays, a lot of apps use CDNs and other cloud providers such as Google (*.bc.googleusercontent.com) or AWS etc. and also querying OCSP for certificates revocations and other stuff which have a lot of IPs resolving to similar dns and are used for milion different reasons.

To give another example to highlight why this feature si needed:

Let's say I want to allow port 80 for a specific IP which resolves to Google cloud (*.bc.googleusercontent.com) and let's say i want to allow it but only once to allow the app to download some stuff it needs from the cloud, but I do not want the app to make (leak) any other http (80) traffic. 

Now this is extremely difficult to set a permanent rule, because there are only 2 options:

> if I permently/temporarilly allow port 80 - then maybe i will leak other http data
> if i allow port 80 only for that IP - then, since this is not a DNS based firewall, i cannot wildcard the domain, thus it is impossible to allow the app each times it connects under different cloud IP (since there are a gazzillion IPs in google cloud) as that would mean to create too many rules. Just clicking 'allow once' is much faster and gets the job done whenever needed.

I hope my example is relevant.

**Comment by fusionneur** [comment]
A representation of the previous example:
![image](https://user-images.githubusercontent.com/7840350/150885219-18a56982-d317-49b6-8dc7-3e1ffa5dd951.png)

**Comment by vuslatx** [comment]
It would be great!!

**Comment by PegHorse** [comment]
Yes, i think it would be great if you could find a way to prevent process Execution until we took our decision, that could prevents errors like that.

**Comment by thisisthelongestusernameyouhaveeverseen** [comment]
Allow until next restart is a good idea
can use 'task scheduler' for it

and also disallow until n time or next restart is missing from Simplewall

**Comment by thisisthelongestusernameyouhaveeverseen** [comment]
Until the process/pid exist is good, but if you want to log the IPs that apps/services want to connect to, then just enable logging! simplewall will log them all in a file for you

**Comment by pbarney** [comment]
I believe @henrypp is closing _all_ issues related to PID identification because of [this comment](https://github.com/henrypp/simplewall/issues/1047#issuecomment-1554743388):

> It cannot be done because of microsoft (as always), not any of FWPM_NET_EVENT have PID information.

If the events that Simplewall receives doesn't contain the PID, then it must take a much more robust solution (like an entirely from-the-ground-up software firewall).
